Renamed bot-twitter to bot

benediktkaiser opened the pull request:
closes #128 

Still TODO: 

- Document all secrets that need to be changed and provide the commands to do so. 


IMPORTANT: Do not merge this PR until secrets have been transfered correctly.

benediktkaiser commented:
I already copied all the secret for localhost from `auction-ui/localhost/bot-twitter` to `auction-ui/localhost/bot`.

Please execute the following commands to also transfer the secrets for `staging` and `production`.

```bash
$ secret export auction-ui/main.auction-ui.k8s.sidestream.tech/bot-twitter | secret import auction-ui/main.auction-ui.k8s.sidestream.tech/bot - 

$ secret export auction-ui/production.auction-ui.k8s.sidestream.tech/bot-twitter | secret import auction-ui/production.auction-ui.k8s.sidestream.tech/bot - 

$ secret export auction-ui/auctions.makerdao.network/bot-twitter | secret import auction-ui/auctions.makerdao.network/bot - 
```

These commands take into account that we have the following secret locations: 

- localhost
- main.auction-ui.k8s.sidestream.tech
- production.auction-ui.k8s.sidestream.tech
- auctions.makerdao.network


If I am missing any please correct me. Thanks!

benediktkaiser commented:
I already copied all the secret for localhost from `auction-ui/localhost/bot-twitter` to `auction-ui/localhost/bot`.

Please execute the following commands to also transfer the secrets for `staging` and `production`.

```bash
$ secret export auction-ui/main.auction-ui.k8s.sidestream.tech/bot-twitter | secret import auction-ui/main.auction-ui.k8s.sidestream.tech/bot - 

$ secret export auction-ui/production.auction-ui.k8s.sidestream.tech/bot-twitter | secret import auction-ui/production.auction-ui.k8s.sidestream.tech/bot - 

$ secret export auction-ui/auctions.makerdao.network/bot-twitter | secret import auction-ui/auctions.makerdao.network/bot - 
```

These commands take into account that we have the following secret locations: 

- localhost
- main.auction-ui.k8s.sidestream.tech
- production.auction-ui.k8s.sidestream.tech
- auctions.makerdao.network


If I am missing any please correct me. Thanks!

valiafetisov commented:
If I search `bot-twitter` within the project, I still get those results:
```
.dockerignore:
  8  
  9: # node/npm (frontend, bot-twitter)
  10  node_modules

README.md:
  10  
  11: Please follow service-level readme instructions inside [frontend](./frontend) and [bot-twitter](./bot-twitter) folders.
  12  

bot/README.md:
  55  - have a certain formatting and linting setup
  56: - don't forget to create `./bot-twitter/.env` file
  57  
```
and
```
bot/package-lock.json:
  1  {
  2:   "name": "auction-twitter-bot",
  3    "version": "1.0.0",

  7      "": {
  8:       "name": "auction-twitter-bot",
  9        "version": "1.0.0",

bot/package.json:
  1  {
  2:   "name": "auction-twitter-bot",
  3    "version": "1.0.0",
```

benediktkaiser commented:
I changed all the names. One last thing. The drone cannot pull the new docker image of the bot (as I renamed it). Do we need to change the name of the bot on the docker registry too? 

See: https://drone.sidestream.tech/sidestream-tech/unified-auctions-ui/606/3/3

valiafetisov commented:
Good question, I would address it to devops team. My guess it that it's an optional step that pulls image cache, that's why it also doesn't affect general CI status. But I personally can't find _any_ auction-ui images in our registry for some reason https://github.com/orgs/sidestream-tech/packages

valiafetisov commented on .drone.yml:
```diff
@@ -176,11 +176,11 @@ data:
         values_to_set:
           - frontend.deployment.tag=$DRONE_COMMIT_SHA
-          - botTwitter.deployment.tag=$DRONE_COMMIT_SHA
+          - bot.deployment.tag=$DRONE_COMMIT_SHA
```
Found one more place that still uses `botTwitter`:
https://github.com/sidestream-tech/unified-auctions-ui/blob/63d643212529df630479c2dbd6f95faf3364852e/kubernetes/helm/templates/aws-chamber-read-secret.yaml#L12

benediktkaiser replied:
Changed it 😅

valiafetisov commented:
> it would re-pull it after the first build

I suppose since this CI step doesn't publish image built in the next step, it will fail till there is an image

sherincheriyan commented:
As @valiafetisov  said, It is trying to pull the image with the main tag which hasn't published yet I think. I am waiting for Nils to pass me the OTP for the sideborg user to verify the image
```
 docker pull ghcr.io/sidestream-tech/unified-auctions-ui/bot:main
Error response from daemon: manifest unknown
```

valiafetisov commented:
@benediktkaiser can you please tick boxes in the issue that you've done?
